src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5411,7 +5411,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
 
       def typedPackageDef(pdef0: PackageDef) = {
         val pdef = treeCopy.PackageDef(pdef0, pdef0.pid, pluginsEnterStats(this, pdef0.stats))
-        val pid1 = typedQualifier(pdef.pid).asInstanceOf[RefTree]
+        val pid1 = context.withMode(ContextMode.InPackageClauseName)(typedQualifier(pdef.pid).asInstanceOf[RefTree])
```

Comment:
We should forward this to 2.13 and use it to fix lookup when type-checking a package clause.

In 2.13:

```
package a.x
class C { def inA = 0 }
```
```
package b
import a._
package x {
  class C { def inB = 0 }
}
```
the `PackageDef.pid` for `b.x` will refer to the package symbol `a.x` because of the new lookup rules.

I couldn't spot any uses of `PackageDef.pid` in the compiler, so it's probably benign.